packages/ethereum-storage/test/ipfs-manager.test.ts

```diff
@@ -379,4 +379,17 @@ describe('Ipfs manager', () => {
     );
     expect(spy).toHaveBeenCalledTimes(5);
   });
+
+  it('should throw on add timeout', async () => {
+    ipfsManager = new IpfsManager({ ...ipfsGatewayConnection, timeout: 1 }, retryTestErrorHandling);
+    const assertionsNb = 10;
+    expect.assertions(assertionsNb);
+    let assertions = [];
+    for (let i = 0; i < assertionsNb; i++) {
+      assertions.push(
+        expect(ipfsManager.add('test')).rejects.toThrowError('Ipfs add request timeout'),
+      );
+    }
+    await Promise.all(assertions);
+  });
```

Comment:
This test was failing before (at least once in the loop of 10 assertions). It is green now.